datafusion/core/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -551,17 +558,46 @@ impl SortPreservingMergeStream {
         if self.aborted {
             return Poll::Ready(None);
         }
+        let num_streams = self.streams.num_streams();
+
+        // Init all cursors and the loser tree in the first poll
+        if self.loser_tree.is_empty() {
+            // Ensure all non-exhausted streams have a cursor from which
```

Comment:
It might be easier to follow if this method were split into a method called `init_loser_tree` with a doc comment explaining what it does